src/test_workflow/integ_test/local_test_cluster.py

```diff
@@ -113,14 +113,12 @@ def wait_for_service(self):
                     return
             except requests.exceptions.ConnectionError:
                 logging.info("Service not available yet")
-                if self.stdout:
+                if self.process_handler.output:
                     logging.info("- stdout:")
-                    with open(os.path.join(self.work_dir, self.stdout.name), "r") as stdout:
-                        logging.info(stdout.read())
-                if self.stderr:
+                    logging.info(self.process_handler.output.read())
```

Comment:
You're checking a property of an instance of `process_handler`, then reaching into a method on that property; this is typically a "wrong interface" smell or at least a missed opportunity for adding a method. Internally you already call `stdout_data = stdout.read`, if no other code uses `.output`, then change it to return `stdout_data` instead of of the `stdout` handle. Same for `.error`. And here you won't have to do the `if`, just `logger.info(self.process_handler.output)`.

Reply:
Thanks for the callout. The situation is a bit tricky. The `stdout_data` is set inside termination logic. However, the read here is when pinging 9200 during service readiness check. I think the confusion is from the name. Maybe I could rename `stdout_data` to `stdout_termination_data`. Let me know if this shall be addressed during this PR or I could work during follow up PR.

Reply:
it can be done later